system-tests/test_example.py

```diff
@@ -0,0 +1,201 @@
+    check_message_pv_name_and_value_type(log_data_second, Value.Value.Double, b'SIM:Spd')
+    check_double_value_and_equality(log_data_second, 5)
+    cons.close()
+
+
+def poll_for_valid_message(consumer):
+    """
+    Polls the subscribed topics by the consumer and checks the buffer is not empty or malformed.
+    :param consumer: The consumer object.
+    :return: The message object received from polling.
+    """
+    msg = consumer.poll()
+    assert not msg.error()
+    return msg
+
+
+def create_consumer():
+    consumer_config = {'bootstrap.servers': 'localhost:9092', 'default.topic.config': {'auto.offset.reset': 'smallest'},
+                       'group.id': uuid.uuid4()}
+    cons = Consumer(**consumer_config)
+    return cons
+
+
+def check_double_value_and_equality(log_data, expected_value):
+    """
+    Initialises the log data object from bytes and checks the union table
+    and converts to Python Double then compares against the expected Double value.
+    :param log_data: Log data object from the received stream buffer
+    :param expected_value: Double value to compare against
+    :return: none
+    """
+    union_double = Double.Double()
+    union_double.Init(log_data.Value().Bytes, log_data.Value().Pos)
+    union_value = union_double.Value()
+    assert math.isclose(expected_value, union_value)
+
+
+def check_message_pv_name_and_value_type(log_data, value_type, pv_name):
+    """
+    Checks the message value name (PV) and value type (type of PV)
+    :param log_data: Log data object from the received stream buffer
+    :param value_type: Flatbuffers value type
+    :param pv_name: Byte encoded string of the PV/channel name
+    :return: none
+    """
+    assert value_type == log_data.ValueType()
+    assert pv_name == log_data.SourceName()
+
+
+def change_pv_value(pvname, value):
+    """
+    Epics call to change PV value provided with the PV name
+    TODO: call inside a docker container rather than locally
+    :param pvname: string PV name
+    :param value: PV value to change to
+    :return: none
+    """
+    try:
+        chan = CaChannel(pvname)
+        print("created CaChannel")
+        chan.searchw()
+        chan.putw(value)
+        chan.getw()
+    except CaChannelException as e:
+        print(e)
+
+
+def check_json_config(json_object, topicname, pvs, schema="f142", channel_provider_type="ca"):
+    """
+    Check the json config is valid that gets sent to the configuration topic.
+    :param json_object: Dictionary containing all config options
+    :param topicname: The data topic name to push updates to
+    :param pvs: The list of PVs to listen for changes
+    :param schema: The flatbuffers schema to check against
+    :param channel_provider_type: Epics v3/v4 specification
+    :return: none
+    """
+    assert json_object["cmd"] == "add"
+    used_pvs = []  # used for checking which PVs have been already configured to avoid setting up multiple streams
```

Comment:
Inline comments should be used sparingly prefer "normal" style comments.

This one is far too long.